protocol/messenger_locker.go

```diff
@@ -0,0 +1,224 @@
+package protocol
```

Comment:
I'd suggest to rename this file to `messenger_maps` because `locker` sounds like some new entity and it never mentioned again inside the file.

Reply:
Yeah, good point, this file used to contain a custom mutex for `Messenger` but was gradually changed to only include custom sync.Maps.